client/src/components/TrailShowPage.js

```diff
@@ -25,15 +34,66 @@ const TrailShowPage = (props) => {
     getTrail()
   }, [])
 
+  const postReview = async (newReviewData) => {
+    if (!userStatus) {
+      setLoginStatus(false)
+    }
+
+    const reviewDataTrailId = { ...newReviewData, trailId: id, userId: props.user.id }
```

Comment:
is this breaking when a user isnt signed in?

Reply:
Also we already have the `trailId` on the backend in the router thanks to merge params so we use that and remove it from this section.  We can also get the user on the backend so we don't need to send `userId` here

Reply:
Hey Fang, I'm about 50% there on this correction but don't think I entirely understand what you're looking for. Do you want us to move this logic to the trailREviewRouter or trailsRouter and access it there using req.user and req.trail?

Reply:
Yep! You are right on track. We can handle this on the server side in the nested router that is handling the creation of a new review.  In your case it looks like the nested `trailReviewRouter`